modules/150-user-authn/hooks/delete_crowd_basic_auth_proxy.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// TODO(ipaqsa): can be deleted after 1.65
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 5},
+	Queue:        "/modules/user-authn/delete-crowd-proxy-ingress",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "crowd-proxy-ingress",
+			ApiVersion: "networking.k8s.io/v1",
+			Kind:       "Ingress",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-user-authn"},
+				},
+			},
```

Comment:
```suggestion
			},
			NameSelector: &types.NameSelector{
				MatchNames: []string{"crowd-basic-auth-proxy"},
			},
```
if I'm not mistaken we could do it this way, making the snapshot of one element specific element, without need to iterate it

Reply:
@miklezzzz we cannot use `NameSelector` and `NamespaceSelector` in the same time in this case because the filtration mechanism uses `OR` between these selectors, and it will find all ingresses in `d8-user-authn` and all ingresses with the `crowd-basic-auth-proxy` name in any namespace. In any way we need to check name to avoid deleting wrong ingress, so i had deleted `NameSelector` and added name checking in the filtration function. 
p.s. We could delete `NamespaceSelector` and use only `NameSelector`, but there is a little little possibility that clients create ingresses with name in their namespaces.